SofaKernel/modules/SofaBaseMechanics/MechanicalObject.h

```diff
@@ -422,8 +417,11 @@ protected :
 
     /// @}
 
-    sofa::core::objectmodel::DataFileName filename;
-    Data< bool> ignoreLoader;
+    //int vsize; ///< Number of elements to allocate in vectors
+    Data< int > d_size;
```

Comment:
We had one too many times where we had no idea if this hidden vsize value was correct, as well as the topology pointer.
The point of making them a proper Data<> and Link<> is to expose then for inspection.
Note that is Sofa we have no reliable way to protect a Data from external changes. The ReadOnly flag is only respected by the GUI, not from other component's code. But this issue is not specific to this Data.